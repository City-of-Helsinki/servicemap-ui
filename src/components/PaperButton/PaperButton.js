import React from 'react';
import PropTypes from 'prop-types';
import { Button, Typography } from '@material-ui/core';
import Container from '../Container';

const PaperButton = ({
  classes, disabled, text, onClick, icon, link,
}) => {
  const clonedIcon = icon ? React.cloneElement(icon, { className: classes.icon }) : null;
  const role = link ? 'link' : 'button';
  return (
    <Container paper className={classes.container}>
      <Button
        classes={{
          label: classes.iconButtonLabel,
        }}
        className={classes.iconButton}
        onClick={onClick}
        role={role}
        disabled={disabled}
      >
        <div className={`${classes.iconContainer} ${disabled ? classes.iconDisabled : ''}`}>
          {clonedIcon}
        </div>
        <Typography variant="body2" className={`${classes.text} ${disabled ? classes.textDisabled : ''}`}>
          {text}
        </Typography>
      </Button>
    </Container>
  );
};

PaperButton.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  disabled: PropTypes.bool,
  icon: PropTypes.node,
  link: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
};

PaperButton.defaultProps = {
  disabled: false,
  icon: null,
  link: false,
  onClick: null,
};


export default PaperButton;
