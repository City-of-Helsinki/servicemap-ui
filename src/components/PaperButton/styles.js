export default theme => ({
  container: {
    flex: '0 0 auto',
    padding: 0,
    width: '88%',
    maxWidth: 356,
  },
  iconButton: {
    flex: '1 0 auto',
    height: 66,
    padding: 0,
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unitDouble,
    justifyContent: 'flex-start',
    textAlign: 'left',
    border: `${theme.palette.detail.alpha} solid 0.5px`,
  },
  iconContainer: {
    display: 'flex',
    flexShrink: 0,
    backgroundColor: theme.palette.primary.main,
    justifyContent: 'center',
    width: 40,
    height: 40,
    borderRadius: '50%',
    margin: theme.spacing.unit,
    marginRight: 11,
    marginLeft: 12,
    boxShadow: `0 4px 8px 0 ${theme.palette.detail.alpha}`,
  },
  icon: {
    color: 'inherit',
    height: 'inherit',
    width: 24,
  },
  iconDisabled: {
    backgroundColor: 'rgba(0, 0, 0, 0.26)',
  },
  text: {
    textTransform: 'none',
    lineHeight: '18px',
  },
  textDisabled: {
    color: 'rgba(0, 0, 0, 0.50)',
  },
});
