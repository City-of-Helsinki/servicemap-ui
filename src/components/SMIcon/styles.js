export default theme => ({
  icon: {
    alignSelf: 'center',
    display: 'inline-block',
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    fontSize: 24,
  },
});
