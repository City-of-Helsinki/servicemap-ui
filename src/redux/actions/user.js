import LocalStorageUtility from '../../utils/localStorage';
import fetchAddress from '../../views/MapView/utils/fetchAddress';

export const setLocale = locale => ({
  type: 'SET_LOCALE',
  locale,
});

export const setInitialLoad = () => ({
  type: 'SET_INITIAL_LOAD',
});

export const setPage = page => ({
  type: 'SET_CURRENT_PAGE',
  page,
});

export const setUserPosition = position => ({
  type: 'SET_POSITION',
  position,
});

const setTheme = theme => ({
  type: 'SET_THEME',
  theme,
});


export const changeLocaleAction = locale => async (dispatch) => {
  dispatch(setLocale(locale));
};

export const actionSetInitialLoad = () => async (dispatch) => {
  dispatch(setInitialLoad());
};

export const setCurrentPage = page => async (dispatch) => {
  dispatch(setPage(page));
};

export const changeTheme = theme => async (dispatch) => {
  LocalStorageUtility.saveItem('theme', theme);
  dispatch(setTheme(theme));
};

export const findUserLocation = () => async (dispatch) => {
  const success = (position) => {
    if (position.coords.accuracy < 10000) {
      fetchAddress({ lat: position.coords.latitude, lng: position.coords.longitude })
        .then((data) => {
          dispatch(setUserPosition({
            coordinates: position.coords,
            allowed: true,
            addressData: data,
          }));
        });
    } else {
      console.warn(`Position accuracy: ${position.coords.accuracy}. Max accuracy: 10000`);
      dispatch(setUserPosition({ coordinates: null, allowed: true, addressData: null }));
    }
  };

  const error = (err) => {
    console.warn(`GeoLocation error:(${err.code}): ${err.message}`);
    if (err.code === 1) {
      dispatch(setUserPosition({ coordinates: null, allowed: false, addressData: null }));
    } else {
      dispatch(setUserPosition({ coordinates: null, allowed: true, addressData: null }));
    }
  };

  navigator.geolocation.getCurrentPosition(success, error);
};


export default { setLocale, changeLocaleAction };
