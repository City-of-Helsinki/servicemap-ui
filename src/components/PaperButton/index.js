import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import PaperButton from './PaperButton';
import styles from './styles';

export default injectIntl(withStyles(styles)(PaperButton));
