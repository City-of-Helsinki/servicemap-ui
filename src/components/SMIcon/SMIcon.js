import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@mui/styles';
import styles from './styles';

const SMIcon = ({
  className, classes, icon, ...rest
}) => (
  <span aria-hidden="true" className={`${className} ${icon} ${classes.icon}`} {...rest} />
);

SMIcon.propTypes = {
  className: PropTypes.string,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  icon: PropTypes.string.isRequired,
};

SMIcon.defaultProps = {
  className: '',
};

export default withStyles(styles)(SMIcon);
