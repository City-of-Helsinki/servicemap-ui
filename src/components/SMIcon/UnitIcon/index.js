import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@mui/styles';
import UnitHelper from '../../../utils/unitHelper';
import styles from './styles';

const UnitIcon = ({
  classes, className, unit, settings,
}) => {
  const iconClass = `${classes.icon} ${className || ''}`;
  if (unit && settings) {
    return <img alt="" src={UnitHelper.getIcon(unit, settings, true)} className={iconClass} aria-hidden="true" />;
  }
  return <img alt="" src={UnitHelper.getDefaultIcon()} className={iconClass} aria-hidden="true" />;
};

UnitIcon.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  className: PropTypes.string,
  unit: PropTypes.objectOf(PropTypes.any),
  settings: PropTypes.objectOf(PropTypes.any),
};

UnitIcon.defaultProps = {
  className: null,
  unit: null,
  settings: null,
};

const mapStateToProps = (state) => {
  const { settings } = state;
  return {
    settings,
  };
};

export default withStyles(styles)(connect(
  mapStateToProps,
)(UnitIcon));
