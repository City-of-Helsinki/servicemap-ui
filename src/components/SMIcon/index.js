import React from 'react';
import SMIcon from './SMIcon';
import defaulMapIcon from '../../assets/images/iconDefaultMap.svg';
import aerialMapIcon from '../../assets/images/iconAerialMap.svg';
import guideMapIcon from '../../assets/images/iconGuideMap.svg';
import locationIcon from '../../assets/images/iconLocation.svg';
import accessibilityIcon from '../../assets/images/iconAccessibility.svg';
import serviceListIcon from '../../assets/images/iconServiceList.svg';
import feedbackIcon from '../../assets/images/iconFeedback.svg';
import helpIcon from '../../assets/images/iconHelp.svg';
import facebookIcon from '../../assets/images/facebook_icon.svg';
import instagramIcon from '../../assets/images/insta_icon.svg';
import pinterestIcon from '../../assets/images/pinterest_icon.svg';
import snapchatIcon from '../../assets/images/snap_icon.svg';
import twitterIcon from '../../assets/images/twitter_icon.svg';
import youtubeIcon from '../../assets/images/youtube_icon.svg';
import vimeoIcon from '../../assets/images/vimeo_icon.svg';

/**
 * Senses
 */
export const HearingIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-hearing-aid" {...rest} />
);

export const ColorblindIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-colour-blind" {...rest} />
);

export const VisualImpairmentIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-visually-impaired" {...rest} />
);

/**
 * Mobility
 */
export const OnFootIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-by-foot" {...rest} />
);

export const ReducedMobilityIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-reduced-mobility" {...rest} />
);

export const StrollerIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-stroller" {...rest} />
);

export const WheelchairIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-wheelchair" {...rest} />
);

export const RollatorIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-rollator" {...rest} />
);
/**
 * Cities
 */
export const HelsinkiIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-coat-of-arms-helsinki" {...rest} />
);

export const EspooIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-coat-of-arms-espoo" {...rest} />
);

export const VantaaIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-coat-of-arms-vantaa" {...rest} />
);

export const KauniainenIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-coat-of-arms-kauniainen" {...rest} />
);

/**
 * General
 */
export const AreaIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-areas-and-districts" {...rest} />
);
export const AddressIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-address" {...rest} />
);

export const MapIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-map-options" {...rest} />
);

export const SearchIcon = ({ ...rest }) => (
  <SMIcon icon="icon-icon-search" {...rest} />
);

// Function to get right icon based on key mapping
export const getIcon = (key, props) => {
  switch (key) {
    case 'foot':
      return <OnFootIcon {...props} />;
    case 'colorblind':
      return <ColorblindIcon {...props} />;
    case 'hearingAid':
      return <HearingIcon {...props} />;
    case 'visuallyImpaired':
      return <VisualImpairmentIcon {...props} />;
    case 'reduced_mobility':
      return <ReducedMobilityIcon {...props} />;
    case 'rollator':
      return <RollatorIcon {...props} />;
    case 'wheelchair':
      return <WheelchairIcon {...props} />;
    case 'stroller':
      return <StrollerIcon {...props} />;

    // Map types
    case 'servicemap':
      return <img aria-hidden alt="" src={defaulMapIcon} {...props} />;
    case 'orthoImage':
      return <img aria-hidden alt="" src={aerialMapIcon} {...props} />;
    case 'guideMap':
      return <img aria-hidden alt="" src={guideMapIcon} {...props} />;

    // Front page buttons
    case 'location':
      return <img aria-hidden alt="" src={locationIcon} {...props} />;
    case 'accessibility':
      return <img aria-hidden alt="" src={accessibilityIcon} {...props} />;
    case 'serviceList':
      return <img aria-hidden alt="" src={serviceListIcon} {...props} />;
    case 'feedback':
      return <img aria-hidden alt="" src={feedbackIcon} {...props} />;
    case 'help':
      return <img aria-hidden alt="" src={helpIcon} {...props} />;

    // Social media links
    case 'facebook':
      return <img aria-hidden alt="" src={facebookIcon} {...props} />;
    case 'instagram':
      return <img aria-hidden alt="" src={instagramIcon} {...props} />;
    case 'pinterest':
      return <img aria-hidden alt="" src={pinterestIcon} {...props} />;
    case 'snapchat':
      return <img aria-hidden alt="" src={snapchatIcon} {...props} />;
    case 'twitter':
      return <img aria-hidden alt="" src={twitterIcon} {...props} />;
    case 'youtube':
      return <img aria-hidden alt="" src={youtubeIcon} {...props} />;
    case 'vimeo':
      return <img aria-hidden alt="" src={vimeoIcon} {...props} />;

    default:
      return null;
  }
};
