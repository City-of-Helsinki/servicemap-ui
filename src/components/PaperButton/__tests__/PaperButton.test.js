// Link.react.test.js
import React from 'react';
import { fireEvent } from '@testing-library/react';
import { initialState } from '../../../redux/reducers/user';
import englishTranslations from '../../../i18n/en';
import PaperButton from '../index';
import { getIcon } from '../../SMIcon';
import { getRenderWithProviders } from '../../../../jestUtils';

const paperButtonProps = {
  messageID: "home.buttons.closeByServices",
  icon: getIcon('location'),
  link: true,
  onClick: () => {},
  subtitleID: "location.notAllowed",
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
  settings: {},
});

describe('<PaperButton />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<PaperButton {...paperButtonProps} />);
    expect(container).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const { container } = renderWithProviders(
      <PaperButton
        {...paperButtonProps}
        onClick={mockCallBack}
      />,
    );
    fireEvent.click(container.querySelector('button'));
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('does render default accessibility attributes correctly', () => {
    const { container } = renderWithProviders(
      <PaperButton messageID="home.buttons.closeByServices" />,
    );
    const buttonBase = container.querySelector('button');
    // Expect role to be link
    expect(buttonBase).toHaveAttribute('role', 'button');
    // Expect tabindex to be 0
    expect(buttonBase).toHaveAttribute('tabindex', '0');
    // Expect aria label
    const ariaLabel = buttonBase.getAttribute('aria-label')
    expect(ariaLabel).toContain(englishTranslations['home.buttons.closeByServices']);
  });

  if('does render given accessibility attributes correctly', () => {
    const { container } = renderWithProviders(
      <PaperButton
        {...paperButtonProps}
        disabled
        aria-label="Test label"
      />,
    );
    const buttonBase = container.querySelector('button');
    // Expect role to be link
    expect(buttonBase).toHaveAttribute('role', 'link');
    // Expect tabindex to be 0
    expect(buttonBase).toHaveAttribute('tabindex', '0');
    // Expect aria label to be given prop
    expect(buttonBase).toHaveAttribute('aria-label', 'Test label');
    // Expect disabled to be true
    expect(buttonBase).toHaveAttribute('disabled', 'true');
  });
});
