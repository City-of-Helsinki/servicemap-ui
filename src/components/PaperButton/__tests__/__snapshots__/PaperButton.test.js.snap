// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaperButton /> should work 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 Container-root-9 Container-margin-11   SM-PaperButton PaperButton-container-1    css-1ps6pg7-MuiPaper-root"
  >
    <button
      aria-label="Show nearby services Location not allowed "
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root PaperButton-iconButton-3 css-1p2nkht-MuiButtonBase-root-MuiButton-root"
      role="link"
      tabindex="0"
      type="button"
    >
      <div
        class="PaperButton-iconContainer-4 "
      >
        <svg
          class="PaperButton-icon-5"
          height="16px"
          version="1.1"
          viewBox="0 0 16 16"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill="inherit"
              fill-rule="nonzero"
              transform="translate(-60.000000, -403.000000)"
            >
              <g
                transform="translate(60.000000, 403.000000)"
              >
                <path
                  d="M12.1463333,4.21533333 C12.1463333,1.909 10.277,0.0396666667 7.97033333,0.0396666667 C5.66366667,0.0396666667 3.79466667,1.909 3.79466667,4.21533333 C3.79466667,7.51966667 7.97033333,12.4093333 7.97033333,12.4093333 C7.97033333,12.4093333 12.1463333,7.50266667 12.1463333,4.21533333 Z M5.911,3.966 C5.911,2.82833333 6.83333333,1.90633333 7.97066667,1.90633333 C9.108,1.90633333 10.0303333,2.82833333 10.0303333,3.966 C10.0303333,5.10366667 9.108,6.02566667 7.97066667,6.02566667 C6.83333333,6.02566667 5.911,5.10366667 5.911,3.966 Z"
                />
                <circle
                  cx="7.97033333"
                  cy="3.966"
                  r="1.01266667"
                />
                <path
                  d="M10.3313333,10.9566667 C9.991,11.4486667 9.67166667,11.877 9.40566667,12.221 C12.0443333,12.3796667 13.7736667,12.9493333 14.417,13.373 C13.6633333,13.87 11.424,14.5686667 7.971,14.5686667 C4.51833333,14.5686667 2.27866667,13.8696667 1.525,13.373 C2.16833333,12.9493333 3.89633333,12.38 6.53233333,12.2213333 C6.26566667,11.8773333 5.946,11.4493333 5.60533333,10.9573333 C2.35833333,11.277 0,12.2376667 0,13.3726667 C0,14.7693333 3.569,15.902 7.971,15.902 C12.3733333,15.902 15.9416667,14.77 15.9416667,13.3726667 C15.9423333,12.237 13.5813333,11.276 10.3313333,10.9566667 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          aria-hidden="true"
          class="MuiTypography-root MuiTypography-body2 PaperButton-text-7 css-1ak20dk-MuiTypography-root"
        >
          Show nearby services
        </p>
        <span
          aria-hidden="true"
          class="MuiTypography-root MuiTypography-caption PaperButton-text-7 css-r3hlsg-MuiTypography-root"
        >
          Location not allowed
        </span>
      </div>
    </button>
  </div>
</div>
`;
